### PR TITLE
Route on unanswered numerics

### PR DESCRIPTION
**Pull request by K9Morphed**
### What is the context of this PR?
Changed various files to enable all number type selectors to show 'Unanswered' as an option and allow Publisher to recognise routing on unanswered numerics where appropriate. Tests written for all number types and are passing.

Trello: https://trello.com/c/wI20lXJW

### How to review 
Have a look over the code and see what you think; check that the tests pass and the product works as expected.

### CODE REVIEW

File: eq-publisher/src/eq_schema/builders/routing2/translateBinaryExpression.js [hunk 1]
```diff
@@ -33,11 +37,16 @@ const buildCheckboxAnswerBinaryExpression = ({ left, right, condition }) => {
 };
 
 const buildBasicAnswerBinaryExpression = ({ left, condition, right }) => {
-  return {
+  let returnVal = {
```
Comment (K9Morphed) [inline eq-publisher/src/eq_schema/builders/routing2/translateBinaryExpression.js]: This should be a const

File: eq-author/src/App/page/Routing/RoutingPage/RoutingEditor/RuleEditor/BinaryExpressionEditor/NumberAnswerSelector.test.js [hunk 2]
```diff
@@ -59,4 +59,26 @@ describe("NumberAnswerSelector", () => {
       customValue: { number: 123 },
     });
   });
+
+  it("should not show the number input field when unanswered is chosen on a ALL type", () => {
```
Comment (SamGodwin2) [inline eq-author/src/App/page/Routing/RoutingPage/RoutingEditor/RuleEditor/BinaryExpressionEditor/NumberAnswerSelector.test.js]:
```suggestion
  it("should not show the number input field when unanswered is chosen on a numeric type", () => {
```
?

File: eq-publisher/src/eq_schema/builders/routing2/translateBinaryExpression.test.js [hunk 3]
```diff
@@ -174,5 +175,28 @@ describe("Should build a runner representation of a binary expression", () => {
         });
       });
     });
+
+    it("can translate unanswered question routing from Author to Runner for all types", () => {
```
Comment (heofs) [inline eq-publisher/src/eq_schema/builders/routing2/translateBinaryExpression.test.js]: Should it be "for all **numeric** types"?

Comment (K9Morphed) [reply eq-publisher/src/eq_schema/builders/routing2/translateBinaryExpression.test.js]: Yeah, that probably makes the most sense. Cheers!

File: eq-author/src/App/page/Routing/RoutingPage/RoutingEditor/RuleEditor/BinaryExpressionEditor/NumberAnswerSelector.js [hunk 4]
```diff
@@ -7,6 +7,16 @@ import { colors, radius } from "constants/theme";
 import { Number, Select, Label } from "components/Forms";
 import VisuallyHidden from "components/VisuallyHidden";
 
+const conditions = {
+  EQUAL: "Equal",
+  NOTEQUAL: "NotEqual",
+  GREATERTHAN: "GreaterThan",
+  LESSTHAN: "LessThan",
+  GREATEROREQUAL: "GreaterOrEqual",
+  LESSOREQUAL: "LessOrEqual",
```
[the remark this answers is not in the document]
Comment (K9Morphed) [reply eq-author/src/App/page/Routing/RoutingPage/RoutingEditor/RuleEditor/BinaryExpressionEditor/NumberAnswerSelector.js]: Done! It makes sense to do what you said anyway, even if there's variations; it does make it much more readable.

**Pull request merged**